lib/matplotlib/backends/_backend_pdf_ps.py

```diff
@@ -196,33 +195,71 @@ def track_glyph(
             The character code within the above subset. If *subset_size* was not
             specified on this instance, then this is just *charcode* unmodified.
         """
-        if self.subset_size != 0:
-            subset = charcode // self.subset_size
-            subset_charcode = charcode % self.subset_size
+        # Normalize for the key.
+        if isinstance(chars, str):
+            charcode = tuple(ord(c) for c in chars)
+        elif not isinstance(chars, tuple):
+            charcode = (chars, )
         else:
+            charcode = chars
+
+        glyph_map = self.glyph_map.setdefault(font.fname, {})
+        key = (charcode, glyph)
+        if key in glyph_map:
+            return glyph_map[key]
+
+        subset_maps = self.used.setdefault(font.fname, [{}])
+        use_next_charmap = False
```

Comment:
I would write this as
```python
use_next_charmap = (
    # Multi-character glyphs always go...
    len(charcodes) > 1
    or (self.subset_size != 0
        # Start filling a new subset...
        and (charcodes[0] >= self.subset_size or charcodes[0] in subset_maps[0]))
)
if use_next_charmap:
    ...
else:
    ...
```
Otherwise it's a bit harder to follow that you first set subset and subset_charcodes, but may later override them if use_next_charmap = True.

Reply:
Sure, done.